woocommerce/woocommerce-ios
# Introduce waiting time track capability

Summary
==========
Partially fix issues #7381 and #7369 by introducing a Waiting time tracker feature that when notified with a `start` and `end` calls, will submit the elapsed time to our Tracks with the provided analytics track. This allows us to measure how long our user waits between doing any action and seeing the result. Right now, this will be used specifically for the Orders performance project, allowing us to measure the waiting time to load the Order Details and Order List views.

The work introduced here is the iOS counterpart of [this Android one](https://github.com/woocommerce/woocommerce-android/pull/7094)

How to Test
==========
1. Launch the app
2. Wait for the My Store to load
3. Go to the Orders section and wait for the Order list to load
4. Select any Order in that list and wait for the Order details to load
5. Verify in the Tracks logs that the following information is showing up:

```
🔵 Tracked dashboard_top_performers_waiting_time_loaded, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("waiting_time"): 4.4139580726623535, AnyHashable("blog_id"): ]
 ```
```
🔵 Tracked dashboard_main_stats_waiting_time_loaded, properties: [AnyHashable("blog_id"):, AnyHashable("is_wpcom_store"): false, AnyHashable("waiting_time"): 5.779787063598633]
```

```
🔵 Tracked order_detail_waiting_time_loaded, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("waiting_time"): 3.745540142059326, AnyHashable("blog_id"):]
```

Update release notes:
